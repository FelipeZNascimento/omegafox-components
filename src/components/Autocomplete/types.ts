export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
