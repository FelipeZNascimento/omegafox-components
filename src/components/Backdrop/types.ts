export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
