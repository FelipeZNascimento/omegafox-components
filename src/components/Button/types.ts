export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
