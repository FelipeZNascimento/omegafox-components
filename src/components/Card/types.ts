export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
