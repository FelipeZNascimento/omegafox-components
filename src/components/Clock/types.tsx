import {
  FOOTBALL_MATCH_STATUS,
  AMERICAN_FOOTBALL_MATCH_STATUS
} from './constants';
import { TBetValues } from '../types';

type TClock = {
  time: number;
};

type FootballKeys = keyof typeof FOOTBALL_MATCH_STATUS;
export type TClockFootball = TClock & {
  status: typeof FOOTBALL_MATCH_STATUS[FootballKeys];
};

type AmericanFootballKeys = keyof typeof AMERICAN_FOOTBALL_MATCH_STATUS;
export type TClockAmericanFootball = TClock & {
  status: typeof AMERICAN_FOOTBALL_MATCH_STATUS[AmericanFootballKeys];
};

export interface IClockProps {
  betValue: TBetValues | null;
  clock: TClockAmericanFootball | TClockFootball;
  isMatchStarted?: boolean;
  isExpandable: boolean;
  isExpanded: boolean;
  isMatchEditable: boolean;
  timestamp: number;
}

export interface IClockMobileProps {
  clock: TClockAmericanFootball | TClockFootball;
  timestamp: number;
}
