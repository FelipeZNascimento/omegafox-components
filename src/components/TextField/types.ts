export interface ITextFieldProps {
  defaultValue?: string;
  description?: string;
  inputName: string;
  isError?: boolean;
  placeholder: string;
  type?: 'password' | 'email' | 'text';
  onBlur?: null | ((e: React.FocusEvent<HTMLInputElement>) => void);
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
}
