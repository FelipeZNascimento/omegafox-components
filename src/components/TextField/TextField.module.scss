@import "../../styling/alignments.scss";
@import "../../styling/colors.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/global.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.container {
  @include font-sized(m);
  @include marged(all, none);
  @include padded(top, none);
  @include padded(bottom, l);

  border: 0;
  display: inline-flex;
  flex-direction: column;
  min-width: 0;
  position: relative;
  vertical-align: top;
  width: 100%;

  &:focus-within label {
    @extend .filled;
  }
  &:focus-within legend {
    @extend .hidden;
  }

  label {
    @include padded(all, none);
    @include font-colored($color-grey4);

    display: block;
    left: 0;
    position: absolute;
    top: 0;
    transform: translate(20px, 16px) scale(1);
    transform-origin: top left;
    transition: color 200ms cubic-bezier(0, 0, 0.2, 1) 0ms, transform 200ms cubic-bezier(0, 0, 0.2, 1) 0ms;
    pointer-events: none;
    z-index: 1;
  }

  .filled {
    @include font-colored($color-grey3);
    @include font-sized(m);

    background-color: inherit;
    max-width: 3000px;
    transition: 200ms cubic-bezier(0, 0, 0.2, 1) 0ms;
    transform: translate(18px, -8px);
  }
}

.inputContainer {
  align-items: center;
  box-sizing: border-box;
  border-radius: $border-radius;
  cursor: text;
  display: inline-flex;
  position: relative;
  width: 100%;

  &Disabled {
    @include bg-rgba-colored($color-grey2, 0.5);
  }

  &:hover fieldset {
    border-width: 2px;
  }

  input {
    @include marged(all, none);
    @include padded(vertical, l);
    @include padded(horizontal, l2);

    width: 100%;
    border: 0;
    display: block;
    min-width: 0;
    box-sizing: content-box;
  }

  fieldset {
    @include marged(all, none);
    @include padded(vertical, none);
    @include padded(horizontal, s);

    border: 1px solid $color-grey3;
    border-radius: $border-radius;
    top: -5px;
    left: 0;
    right: 0;
    bottom: 0;
    overflow: hidden;
    position: absolute;
    pointer-events: none;

    legend {
      @include padded(all, none);
      @include aligned(left);

      height: 11px;
      display: block;
      font-size: 0.75em;
      max-width: 0.01px;
      transition: max-width 50ms cubic-bezier(0, 0, 0.2, 1) 0ms;
      visibility: hidden;
      width: auto;
    }

    .hidden {
      @include font-sized(m);
      @include padded(horizontal, s);
      max-height: 16px;
      max-width: 3000px;
    }
  }

  .error {
    border: 2px solid $color-qatar-red;
  }
}

.description {
  @include font-colored($color-grey4);
  @include font-sized(s);
  @include padded(top, s3);
  @include padded(horizontal, l2);
}
