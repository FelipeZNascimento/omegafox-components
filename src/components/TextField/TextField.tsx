import React, { useEffect, useState } from 'react';

import classNames from 'classnames';
import { ITextFieldProps } from './types';
import styles from './TextField.module.scss';

export const TextField = ({
  defaultValue = '',
  description = '',
  inputName,
  isError = false,
  placeholder,
  type = 'text',
  onBlur,
  onChange
}: ITextFieldProps) => {
  const [value, setValue] = useState('');

  useEffect(() => {
    setValue(defaultValue);
  }, [defaultValue]);

  const legendClass = classNames({
    [styles.hidden]: value !== ''
  });

  const filledClass = classNames({
    [styles.filled]: value !== ''
  });

  const fieldsetClass = classNames({
    [styles.error]: isError
  });

  const handleBlur = (e: React.FocusEvent<HTMLInputElement>) => {
    if (onBlur) {
      onBlur(e);
    }
  };

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setValue(e.target.value);
    onChange(e);
  };

  return (
    <div className={styles.container}>
      <label className={filledClass} htmlFor={inputName}>
        {placeholder}
      </label>
      <div className={styles.inputContainer}>
        <input
          defaultValue={defaultValue}
          id={inputName}
          name={inputName}
          type={type}
          onBlur={handleBlur}
          onChange={handleChange}
        />
        <fieldset className={fieldsetClass}>
          <legend className={legendClass}>{placeholder}</legend>
        </fieldset>
      </div>
      {description && <div className={styles.description}>{description}</div>}
    </div>
  );
};
