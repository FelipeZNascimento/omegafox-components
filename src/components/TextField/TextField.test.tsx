import React from 'react';
import { render } from '@testing-library/react';
import { TextField } from './TextField';
import { ITextFieldProps } from './types';

describe('<InputField />', () => {
  const renderComponent = ({ inputName, placeholder, type }: ITextFieldProps) =>
    render(
      <TextField
        inputName={inputName}
        placeholder={placeholder}
        type={type}
        onchange={jest.fn()}
      />
    );

  it('should render', () => {
    const { container } = renderComponent({
      defaultValue: '',
      isError: false,
      inputName: 'inputName',
      placeholder: 'Placeholder',
      onchange: jest.fn
    });

    expect(container).toMatchSnapshot();
  });
});
