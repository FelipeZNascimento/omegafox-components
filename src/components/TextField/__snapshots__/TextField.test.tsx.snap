// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> should render 1`] = `
<div>
  <div
    class="container"
  >
    <label
      class=""
      for="inputName"
    >
      Placeholder
    </label>
    <div
      class="inputContainer"
    >
      <input
        id="inputName"
        name="inputName"
        type="text"
        value=""
      />
      <fieldset
        class=""
      >
        <legend
          class=""
        >
          Placeholder
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
